fbfmaproom/fbfmaproom.py

```diff
@@ -106,14 +105,60 @@ def units(self):
             self._units = self.open().attrs.get('units')
         return self._units
 
+    def open(self, val_min=None, val_max=None):
+        path = data_path(self.path)
+        try:
+            ds = xr.open_zarr(path, consolidated=False)
+        except Exception as e:
+            raise Exception(f"Couldn't open {path}") from e
+        ds = ds.rename({
+            v: k
+            for k, v in self.var_names.items()
+            if v is not None and v != k
+        })
+        da = ds["value"]
+
+
+        if val_min is None:
+            val_min = self.range[0]
+        if val_max is None:
+            val_max = self.range[1]
+        da.attrs["colormap"] = self.colormap
+        da.attrs["scale_min"] = val_min
+        da.attrs["scale_max"] = val_max
+        return da
```

Comment:
I'm kind of confused what's going on here, I thought we were using this class you designed to store attributes, or is the class more a descriptor for a zarr file (set) (as opposed to the xarray data set)?

Reply:
We had two different functions for opening datasets depending on whether they were forecasts or obs. Then we had if/then/else scattered all over the place to deal with the fact that we never know which we'll be dealing with. What I'm doing here is trying to centralize the differences in one place so that most of the code doesn't need to care if a given set of numbers comes from a forecast or an obs.

The new class doesn't contain any data, we still use xarray for that. It's just an abstraction that helps with opening a zarr dataset and getting xarray out of it.

I'm not actually finding myself entirely convincing here. Have to think more explicitly about the role of each of the classes.

Reply:
The question I was asking myself is whether the differences between ObsDataset and ForecastDataset are essential differences, or if they're just historical accidents and could be merged into one. After a closer look, I think some of them are incidental but the differences in the `select` method, and the differences between `ForecastDataset.select` and `xr.DataArray.sel`, are significant.

There might be simplifications to be made, but I'd like to go ahead with this for now if you have no further comments @kgraaf .